library/ssl_tls13_client.c

```diff
@@ -1183,11 +1183,11 @@ int mbedtls_ssl_tls13_write_client_hello_exts( mbedtls_ssl_context *ssl,
             return( ret );
         p += ext_len;
 
-        /* Initializes the status to `indication sent`. It will be updated to
-         * `accepted` or `rejected` depending on whether the EncryptedExtension
-         * message will contain an early data indication extension or not.
+        /* Initializes the status to `rejected`. It will be updated to
+         * `accepted` if the EncryptedExtension message contain an early data
+         * indication extension.
          */
-        ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_INDICATION_SENT;
+        ssl->early_data_status = MBEDTLS_SSL_EARLY_DATA_STATUS_REJECTED;
```

Comment:
- Rename `early_data_status`, to show that's not for outside
- For internal usage, I prefer not reuse `MBEDTLS_SSL_EARLY_DATA_STATUS_REJECTED` . Although I also think it can be reuse. 
- And this information can be represent by `handshake->sent_extensions & MBEDTLS_SSL_EXT_MASK( EARLY_DATA ) ` .  :)
```suggestion
        ssl->early_data_state = MBEDTLS_SSL_EARLY_DATA_STATUS_ALLOW_SEND;
```

Reply:
`early_data_status` is definitively for the user, see https://github.com/Mbed-TLS/mbedtls/pull/6621#discussion_r1027718541.